Makefile.in

```diff
@@ -515,7 +515,7 @@ endif
 
 $(objroot)lib/$(LIBJEMALLOC).$(SOREV) : $(if $(PIC_CFLAGS),$(C_PIC_OBJS),$(C_OBJS)) $(if $(PIC_CFLAGS),$(CPP_PIC_OBJS),$(CPP_OBJS))
 	@mkdir -p $(@D)
-	$(CC) $(DSO_LDFLAGS) $(call RPATH,$(RPATH_EXTRA)) $(LDTARGET) $+ $(LDFLAGS) $(LIBS) $(EXTRA_LDFLAGS)
+	$(CXX) $(DSO_LDFLAGS) $(call RPATH,$(RPATH_EXTRA)) $(LDTARGET) $+ $(LDFLAGS) $(LIBS) $(EXTRA_LDFLAGS)
```

Comment:
@davidtgoldblatt could certainly use your eyes here -- it looks the main difference is g++ links libstdc++? Any concerns around that?

Reply:
@interwq We have to ensure unwinder symbols are found. If there is a case _not_ to use g++ in general case, at least it should be used on MacOS (so use `#ifdef __APPLE__` macro).

Reply:
What we do currently certainly seems like a bug when compiling in C++ mode. It seems like we should link with $(CXX) when we're compiling with C++ support and $(C) otherwise, right? I think we want to support environments without a C++ compiler, but we should definitely not use $(C) to compile a library with hard C++ dependencies.

Reply:
@davidtgoldblatt On MacOS my suggestion works just fine (and is a repeated case with a number of ports). Fix was suggested here: https://trac.macports.org/ticket/65945#comment:14

I cannot be sure of other systems. If we have to consider any non-C++ setup, please tell me how to modify the fix, and I gonna update the PR accordingly.

Reply:
@barracuda156 yes I'd like to switch to CXX here too. Likely David mentioned, we should use `$(CXX)` when the C++ support is enabled, which is specified in this flag:
https://github.com/jemalloc/jemalloc/blob/dev/Makefile.in#L317

Reply:
@interwq @davidtgoldblatt Sorry for a delay. So should I just make it like this?
```
ifeq (@enable_cxx@, 1)
    $(CXX) $(DSO_LDFLAGS) $(call RPATH,$(RPATH_EXTRA)) $(LDTARGET) $+ $(LDFLAGS) $(LIBS) $(EXTRA_LDFLAGS)
else
    $(CC) $(DSO_LDFLAGS) $(call RPATH,$(RPATH_EXTRA)) $(LDTARGET) $+ $(LDFLAGS) $(LIBS) $(EXTRA_LDFLAGS)
endif
```
(I am not good with autotools makefiles.)